pkgs/applications/misc/input-leap/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, fetchFromGitHub, cmake, curl, xorg, avahi, qtbase, qttools, libei ? null, libportal,
+  mkDerivation, pkg-config, openssl, wrapGAppsHook,
+  avahiWithLibdnssdCompat ? avahi.override { withLibdnssdCompat = true; },
+  fetchpatch
+}:
+
+mkDerivation rec {
+  pname = "input-leap";
+  version = "unstable-2023-05-24";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
```

Comment:
```suggestion
    owner = "input-leap";
    repo = "input-leap";
```

Reply:
I'm a bit confused as to why we should duplicate this string literal instead of using the name it already has — would you care to offer some explanations for your less obvious suggestions?

Reply:
1. Things that will never change should not be parameterized. 

2. There is no real benefit of using `pname` as a parameter for other attributes

This is not like `version`, that will change after every new release.

3. `pname` is not always useful as parameter. 

Examples:

- openmsx != openMSX
- bochs-emu != bochs
- Bochs != bochs
- TECO != teco
- Imaging != pil (I am still waiting for a function that transforms pil into Imaging!)

4. Using a useless parameterization creates a dependency link in the evaluation graph of the code, complicating its static evaluation analisys for nothing besides a dubious aesthetics.

Reply:
Amusingly — this reference (I'm hesitant to call it a parameter, it's just a variable) came from the Barrier derivation originally, so it actually saved me some edits when indeed I changed the `pname` 😁

But given that I don't anticipate it changing frequently, I don't really mind if you prefer a literal string here aesthetically.